spec/my_enumerables_spec.rb

```diff
@@ -0,0 +1,226 @@
+      expect(%w[ant bear cat].my_all?(/t/)).to eq(%w[ant bear cat].all?(/t/))
+    end
+  end
+
+  context '#my_any' do 
+    it 'returns true when at least one item matches given condition' do
+      expect(%w[ant bear cat].my_any? { |word| word.length >= 3 }).to be true
+    end
+    it 'returns false when no item matches given condition' do
+      expect(%w[ant bear cat].my_any? { |word| word.length >= 5 }).to be false
+    end
+
+    it 'returns true when no block given and at least one item neither nil or false' do
+      expect([nil, true, false].my_any?).to eq true
+    end
+
+    it 'returns false when given an empty array (at least one item being neither nil or false)' do
+      expect([].my_any?).to eq false
+    end
+
+    it 'returns false when no item matches a given pattern' do
+      expect(%w[ant bear cat].my_any?(/d/)).to eq false
+    end
+
+    it 'all above-mentioned tests match the official enumerable method counterpart' do
+      expect(%w[ant bear cat].my_any? { |word| word.length >= 3 }).to eq(%w[ant bear cat].any? { |word| word.length >= 3 })
+      expect(%w[ant bear cat].my_any? { |word| word.length >= 5 }).to eq(%w[ant bear cat].any? { |word| word.length >= 5 })
+      expect([nil, true, false].my_any?).to eq([nil, true, false].any?)
+      expect([].my_any?).to eq([].any?)
+      expect(%w[ant bear cat].my_any?(/d/)).to eq(%w[ant bear cat].any?(/d/))
+    end
+  end 
+
+  context '#my_none' do 
+    it 'returns false when all items match a given condition' do
+      expect(%w[ant bear cat].my_none? { |word| word.length >= 3 }).to be false
+    end
+    it 'returns true when all items do not match a given condition' do
+      expect(%w[ant bear cat].my_none? { |word| word.length >= 5 }).to be true
+    end
+
+    it 'returns false when no block given and all items are neither nil or false' do
+      expect([nil, true, false].my_none?).to eq false
+    end
+
+    it 'returns true when no block given with an empty array' do
+      expect([].my_none?).to eq true
+    end
+
+    it 'returns true when all items do not match a given pattern' do
+      expect(%w[ant bear cat].my_none?(/d/)).to eq true
+    end
+
+    it 'all above-mentioned tests match the official enumerable method counterpart' do
+      expect(%w[ant bear cat].my_none? { |word| word.length >= 3 }).to eq(%w[ant bear cat].none? { |word| word.length >= 3 })
+      expect(%w[ant bear cat].my_none? { |word| word.length >= 5 }).to eq(%w[ant bear cat].none? { |word| word.length >= 5 })
+      expect([nil, true, false].my_none?).to eq([nil, true, false].none?)
+      expect([].my_none?).to eq([].none?)
+      expect(%w[ant bear cat].my_none?(/d/)).to eq(%w[ant bear cat].none?(/d/))
+    end
+  end
+  
+  context '#my_count' do
+    it 'returns array length when no block given' do
+      expect([1, 2, 3, 4].my_count).to eq 4
+    end
+
+    it 'returns number of items in array when item is an argument' do 
```

Comment:
it returns the number of items that match the argument

Reply:
Please refer to Commit No. #6